crates/external_services/src/grpc_client.rs

```diff
@@ -0,0 +1,238 @@
+    ) -> DRResult<CalSuccessRateResponse> {
+        let params = dynamic_routing_config
+            .params
+            .map(|vec| {
+                if !vec.is_empty() {
+                    vec.into_iter()
+                        .map(|param| param.to_string())
+                        .collect::<Vec<String>>()
+                        .join(":")
+                } else {
+                    String::default()
+                }
+            })
+            .get_required_value("Vector of params")
+            .change_context(DRError::ClientBuildingFailed)?;
+
+        let labels = label_input
+            .into_iter()
+            .map(|conn_choice| conn_choice.to_string())
+            .collect::<Vec<_>>();
+
+        let config = dynamic_routing_config.config.map(ForeignFrom::foreign_from);
+
+        let request = tonic::Request::new(CalSuccessRateRequest {
+            id: id.get_string_repr().to_owned(),
+            params,
+            labels,
+            config,
+        });
+
+        let mut client = self.clone();
+
+        let response = client
+            .fetch_success_rate(request)
+            .await
+            .change_context(DRError::GrpcServerResponseFailure)?
+            .into_inner();
+
+        Ok(response)
+    }
+
+    async fn update_success_rate(
+        &self,
+        id: id_type::ProfileId,
+        dynamic_routing_config: DynamicRoutingConfig,
+        label_input: Vec<RoutableConnectorChoiceWithStatus>,
+    ) -> DRResult<UpdateSuccessRateWindowResponse> {
+        let config = dynamic_routing_config.config.map(ForeignFrom::foreign_from);
+
+        let labels_with_status = label_input
+            .into_iter()
+            .map(|conn_choice| LabelWithStatus {
+                label: conn_choice.routable_connector_choice.to_string(),
+                status: conn_choice.status,
+            })
+            .collect();
+
+        let params = dynamic_routing_config
+            .params
+            .map(|vec| {
+                if !vec.is_empty() {
+                    vec.into_iter()
+                        .map(|param| param.to_string())
+                        .collect::<Vec<String>>()
+                        .join(":")
+                } else {
+                    String::default()
+                }
+            })
+            .get_required_value("Vector of params")
+            .change_context(DRError::ClientBuildingFailed)?;
+
+        let request = tonic::Request::new(UpdateSuccessRateWindowRequest {
+            id: id.get_string_repr().to_owned(),
+            params,
+            labels_with_status,
+            config,
+        });
+
+        let mut client = self.clone();
+
+        let response = client
+            .update_success_rate_window(request)
+            .await
+            .change_context(DRError::GrpcServerResponseFailure)?
+            .into_inner();
+
+        Ok(response)
+    }
+}
+
+impl ForeignFrom<CurrentBlockThreshold> for DynamicCurrentThreshold {
```

Comment:
Why can't we have a traditional `From` here?

Reply:
Cause of the orphan rule, since ` CurrentBlockTreshold` is in another crate